### PR TITLE
Add basic unit tests for CTYP-239

### PR DESCRIPTION
**Pull request by kazuhiro**

### CODE REVIEW

File: module-check/src/test/clojure/clojure/core/typed/test/analyze_test.clj [hunk 1]
```diff
@@ -1,4 +1,16 @@
 (ns clojure.core.typed.test.analyze-test
-  (:require [clojure.core.typed.analyze-clj :as ana]))
+  (:require [clojure.test :refer :all]
+            [clojure.core.typed.analyze-clj :as ana]))
 
 (let [{:keys [a]} nil])
```
Comment (frenchy64) [inline module-check/src/test/clojure/clojure/core/typed/test/analyze_test.clj]: Please remove this line if you're going to reuse this file.

File: module-check/src/test/clojure/clojure/core/typed/test/analyze_test.clj [hunk 2]
```diff
@@ -1,4 +1,16 @@
...
+
+(deftest analyze-test
+  (testing "ast-for-file-for-cljc"
+    (is
```
Comment (frenchy64) [inline module-check/src/test/clojure/clojure/core/typed/test/analyze_test.clj]: `(is (ana/ast-for-file ...))` will do here.

**Comment by frenchy64** [comment]
Actually, I just want a tests that that use check-ns. Don't unit test any of the implementation details.

**Comment by kazuhiro** [comment]
As `check-ns` is not tested only in case of `*.cljc` files, I removed these detailed tests in favour of simple `check-ns` one.

File: module-check/src/test/resources/clojure/core/typed/test/dummy_cljc.cljc [hunk 3]
```diff
@@ -0,0 +1,2 @@
+(ns clojure.core.typed.test.dummy_cljc
```
Comment (frenchy64) [inline module-check/src/test/resources/clojure/core/typed/test/dummy_cljc.cljc]: This underscore should be a dash.

**Comment by kazuhiro** [comment]
Is there anything I could do to help with CTYP-239?

**Comment by frenchy64** [comment]
We need to test the situation where there is both a .clj and .cljc of the same name present, and it should pick the .clj. I suggest adding a type error to the .cljc file which never gets loaded.

**Comment by kazuhiro** [comment]
I added a test case with this scenario, however, I choose throwing an exception instead of type error.

**Comment by frenchy64** [comment]
Please change the body to a type error like `(inc 'a)`. Some future version of core.typed might not implicitly evaluate its forms after type checking.

After, please squash into one commit and attach the .patch to the corresponding JIRA issue.

Thanks!

**Comment by kazuhiro** [comment]
Done. Patch attached in jira :)

**Pull request closed without merging**